mypy/checkexpr.py

```diff
@@ -2431,6 +2431,10 @@ def has_member(self, typ: Type, member: str) -> bool:
         """Does type have member with the given name?"""
         # TODO: refactor this to use checkmember.analyze_member_access, otherwise
         # these two should be carefully kept in sync.
+        if isinstance(typ, TypeVarType):
+            typ = typ.upper_bound
```

Comment:
Could you also add a test case also for this branch? (Similar to the one you already have.)